File: competitive_programming/codeforces/07_typescript/282a_bit_plus_plus.ts
function bitPlusPlus(operations: string[]): number {
  let value = 0;
  for (const operation of operations) value += operation.includes("++") ? 1 : -1;
  return value;
}

if (bitPlusPlus(["++X", "X++", "--X"]) !== 1) throw new Error("bit++ failed");
if (bitPlusPlus(["X++", "X++", "X++", "X--"]) !== 2) throw new Error("bit++ failed");
console.log("282A bit++ ok");

File: competitive_programming/codeforces/11_golang/282a_bit_plus_plus.go
package main

import (
	"fmt"
	"strings"
)

func bitPlusPlus(operations []string) int {
	value := 0
	for _, operation := range operations {
		if strings.Contains(operation, "++") {
			value++
		} else {
			value--
		}
	}
	return value
}

func main() {
	if bitPlusPlus([]string{"++X", "X++", "--X"}) != 1 || bitPlusPlus([]string{"X++", "X++", "X++", "X--"}) != 2 {
		panic("bit++ failed")
	}
	fmt.Println("282A bit++ ok")
}
